### PR TITLE
Feature/autocomplete table view

### PR DESCRIPTION
**Pull request by frankboamps**
Hello,

Could you help me review my code and implementation of the autocomplete table view for the first page view controller.

A short demonstration of how the implementation works is below:

Video Walkthrough

### CODE REVIEW

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 1]
```diff
@@ -88,6 +101,61 @@ - (void)viewDidLoad {
...
     [self createButton];
+    [self createFilterForGMSAutocomplete];
+    [self createAutocompleteTableView];
+    self.autocompleteTableView.delegate = self;
+    self.autocompleteTableView.dataSource = self;
+}
+
+#pragma mark - GMSAutocompleteFetcherDelegate
+
+NSMutableArray *resultsArr;
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m]: Why is this here, rather than as a top-level declaration?

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 2]
```diff
@@ -88,6 +101,61 @@ - (void)viewDidLoad {
...
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return 1;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return resultsArr.count;
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m]: How big can this be? Should we check to make sure this isn't too big or are there API-side checks for that?

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 3]
```diff
@@ -88,6 +101,61 @@ - (void)viewDidLoad {
...
+}
+
+- (NSIndexPath *)tableView:(UITableView *)tableView willSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    AutocompleteTableViewCell*cell = [tableView cellForRowAtIndexPath:indexPath];
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m]: Space.
`AutocompleteTableViewCell *cell`
Also, does this not give you a compile-time error?

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 4]
```diff
@@ -88,6 +101,61 @@ - (void)viewDidLoad {
...
+    self.placesSearchBar.text = cell.textLabel.text;
+    //[self.autocompleteTableView reloadData];
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m]: Remove

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 5]
```diff
@@ -143,46 +211,76 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
 
 #pragma mark - UISearchBar delegate method
 
-- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self.autocompleteTableView reloadData];
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m:214]: Do you need to do this here, since you do it later?

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 6]
```diff
@@ -143,46 +211,76 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
...
     if (searchText.length != 0) {
-        //TODO(Franklin): place API stuff like autocomplete here
-        self.userSpecifiedPlaceToVisit = searchText;
+        [self.autocompleteTableView reloadData];
+        [_fetcher sourceTextHasChanged:searchText];
+        [self.view addSubview:self.autocompleteTableView];
+        [self.autocompleteTableView reloadData];
```
Comment (ArunavhaChanda) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m:218]: Why have so many calls to this instead of doing all your work and then having one call to reload at the end of the method? Maybe I'm missing something, so feel free to clarify!

File: Travel Scheduler/Views/AutocompleteTableViewCell.m [hunk 7]
```diff
@@ -0,0 +1,24 @@
+//
+//  AutocompleteTableViewCell.m
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/AutocompleteTableViewCell.m:2]: Do we need this file?

File: Travel Scheduler/Views/AutocompleteTableViewCell.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
+//
+//  AutocompleteTableViewCell.h
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/AutocompleteTableViewCell.h:2]: Do we need this?

File: Travel Scheduler/Views/SlideMenuTableViewCell.h [hunk 9]
```diff
@@ -0,0 +1,19 @@
+//
+//  SlideMenuTableViewCell.h
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/SlideMenuTableViewCell.h:2]: Do we need?

File: Travel Scheduler/Views/SlideMenuTableViewCell.m [hunk 10]
```diff
@@ -0,0 +1,28 @@
+//
+//  SlideMenuTableViewCell.m
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/SlideMenuTableViewCell.m:2]: Do we need?

File: Travel Scheduler/Views/SlideMenuUIView.h [hunk 11]
```diff
@@ -0,0 +1,26 @@
+//
+//  SlideMenuUIView.h
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/SlideMenuUIView.h:2]: This is related to a separate feature. Please don't have spillover code across pull requests!

File: Travel Scheduler/Views/SlideMenuUIView.m [hunk 12]
```diff
@@ -0,0 +1,90 @@
+//
+//  SlideMenuUIView.m
```
Comment (ArunavhaChanda) [inline Travel Scheduler/Views/SlideMenuUIView.m:2]: This is related to a separate feature. Please don't have spillover code across pull requests!

**Review by ArunavhaChanda — CHANGES_REQUESTED** [review]
I really like the UI and demo. Looks good.

However, some changes needed, which would be great to fix/clarify.

Finally, it's best if formatting changes in other files are put in a separate pull request as opposed to a feature one.

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 6 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m:218]: This is mainly due to the fact that when the user wants to change the location and begins to edit the search bar we want the table view to be up to date with what the user has already entered and the other reload data is for the first instance when the user begins to type and table view is created.

This is mainly because of the animations and how we created them to slide in once the user touched the search bar.

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 1 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m]: Moved to the top as a property

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 2 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m]: Since my filter is mainly cities and all roads and other streets are filtered out, I think we should not be concerned. Also, the filter considers bounds given. I currently set it to be centered at Australia but I can change that to SF and also this would be helpful in the different autocomplete I am currently working on.

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 3 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m]: Oops. Didn't catch that.

Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m]: It worked though but it's not the write implementation. Thanks for pointing that out

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 4 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m]: Done

File: Travel Scheduler/View Controllers/FirstScreenViewController.m (hunk 6 above)
Comment (frankboamps) [reply Travel Scheduler/View Controllers/FirstScreenViewController.m:218]: Honestly, After rethinking through the process the reloadData method call should not be there since I made the first call on line 216. I have removed that.

File: Travel Scheduler/Views/SlideMenuUIView.h (hunk 11 above)
Comment (frankboamps) [reply Travel Scheduler/Views/SlideMenuUIView.h:2]: Sure....Taken notice of this

File: Travel Scheduler/Views/AutocompleteTableViewCell.h (hunk 8 above)
Comment (frankboamps) [reply Travel Scheduler/Views/AutocompleteTableViewCell.h:2]: Sorry for inconvenience. Not needed.

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 13]
```diff
@@ -88,6 +100,59 @@ - (void)viewDidLoad {
...
+{
+    NSString *errorMessage = [NSString stringWithFormat:@"%@", error.localizedDescription];
+    NSLog(@"%@", errorMessage);
+}
+
+#pragma mark - Autocomplete Delegate & TabeView DataSource Method
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    static NSString *cellIdentifier = @"AutocompleteTableViewCell";
```
Comment (gilemos) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m:129]: maybe you could put the name directly instead of making a static string?

File: Travel Scheduler/View Controllers/FirstScreenViewController.m [hunk 14]
```diff
@@ -143,46 +208,75 @@ - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
 
 #pragma mark - UISearchBar delegate method
 
-- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self.autocompleteTableView reloadData];
     if (searchText.length != 0) {
-        //TODO(Franklin): place API stuff like autocomplete here
-        self.userSpecifiedPlaceToVisit = searchText;
```
Comment (gilemos) [inline Travel Scheduler/View Controllers/FirstScreenViewController.m:149]: I will need this userSpecifiedPlaceToVisit to create the hub lol
